tools_and_services/setting_up_windows_for_rails.md

```diff
@@ -0,0 +1,145 @@
+# Setting up your Windows 11 machine for running rails projects
+
+## Prerequisites
+
+- Windows 11
+- VS Code
+- SSH
+
+## Install Ubuntu in wsl
+
+Open power shell and run the following command:
+
+```powershell
+wsl --install -d Ubuntu
+```
+
+The command will eventually will tell you to restart your machine. Do that.
+
+When the machine has been restarted it will continue the installation.
+
+It will prompt you for:
+
+- A user name for the linux vm user
+- A password for the linux vm user
+
+When that's in place you will be launched into a linux terminal.
+
+## Install ZSH and Oh-my-zsh (optional)
+
+```sh
+sudo apt -y install zsh
+chsh -s $(which zsh)
+```
+
+Open a new terminal window:
+
+1. Press the "⋁" button in the top of the terminal window. Then select "Ubuntu".
+2. Press "0" to setup zsh with an empty `.zshrc`-file
+3. Run `sh -c "$(curl -fsSL https://raw.githubusercontent.com/ohmyzsh/ohmyzsh/master/tools/install.sh)"`
+
+Done
+
+## Install asdf for zsh
+
+Install required packages for asdf and ruby-build (which is used by the asdf ruby plugin):
+
+```sh
+sudo apt update # Ensure that you have the latest package metadata
+sudo apt -y install curl git curl autoconf bison build-essential libssl-dev libyaml-dev libreadline6-dev zlib1g-dev libncurses5-dev libffi-dev libgdbm6 libgdbm-dev libdb-dev libpq-dev
+```
+
+```sh
+git clone https://github.com/asdf-vm/asdf.git ~/.asdf --branch v0.14.0
+echo '. "$HOME/.asdf/asdf.sh"' >> ~/.zshrc
+source ~/.zshrc # Reload the zshrc configuration for the current session
```

Comment:
I realize that this step requires zsh though zsh is optional.

I don't want to make the guide more complex than necessary, so I wonder if I should leave out the zsh part or make it required.

On the one hand zsh is unnecessary, on the other hand I think it's important to make the terminal experience as immediately user friendly as possible. 🤔

Reply:
I changed it not be optional